### PR TITLE
Open-source 'src/ya-comiste-react-native'

### PR DESCRIPTION
**Pull request by mrtnzlml**
¿Ya comiste? is a side-project I am working on in my free time. It's a React Native application powered by SDUI (Server Driven UI), FBT for translations and Relay for data-fetching. Server is currently not needed but it's already being baked behind the scenes.

### CODE REVIEW

File: src/ya-comiste-react-native/index.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import ExploreDetail from './src/entrypoints/ExploreDetail';
+import ExploreMap from './src/entrypoints/ExploreMap';
+import ExploreMore from './src/entrypoints/ExploreMore';
+import Profile from './src/entrypoints/Profile';
+import Publish from './src/entrypoints/Publish';
+
+// https://account.mapbox.com/
+MapboxGL.setTelemetryEnabled(false);
+MapboxGL.setAccessToken(
+  'pk.eyJ1IjoibXJ0bnpsbWwiLCJhIjoiY2tocXRjODYxMGJlZTJ4bWoyeGZuanZmZiJ9.wX9dThRRU1-80f0UbYbF3w',
```
Comment (tbergq) [inline src/ya-comiste-react-native/index.js:16]: Should this be checked into the repo?

File: src/ya-comiste-react-native/index.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                  component: {
+                    name: 'com.yaComiste.Explore',
+                  },
+                },
+              ],
+              options: {
+                bottomTab: {
+                  text: 'Explore',
+                  // icon: require('./assets/icon-explore.png'),
+                  // icon: 'icon-explore',
```
Comment (tbergq) [inline src/ya-comiste-react-native/index.js:66]: Why these comments?

File: src/ya-comiste-react-native/index.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                  component: {
+                    id: 'PROFILE_SCREEN',
+                    name: 'com.yaComiste.Profile',
+                  },
+                },
+              ],
+              options: {
+                bottomTab: {
+                  text: 'Profile',
+                  // badge: '1',
```
Comment (tbergq) [inline src/ya-comiste-react-native/index.js:115]: Can this be deleted?

File: src/ya-comiste-react-native/index.js (hunk 1 above)
Comment (mrtnzlml) [reply src/ya-comiste-react-native/index.js:16]: I think so, yes - it's a public token.

File: src/ya-comiste-react-native/src/BlockRenderer.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+type Props = {|
+  +block: BlockRendererFragment,
+  +componentId: string,
+|};
+
+function BlockRenderer(props) {
+  const block = props.block.block;
+  let Component = null;
+
+  // TODO: `@match` & `@module` 3D rendering (?)
```
Comment (tbergq) [inline src/ya-comiste-react-native/src/BlockRenderer.js:20]: I don't think that would work for react-native, since all assets have to be in the bundle.

Comment (mrtnzlml) [reply src/ya-comiste-react-native/src/BlockRenderer.js:20]: That is true, I was thinking the same. But the concept is still valid and you can use the same techniques. The only difference is that you are not going to be lazy loading the components over the network, no? You can still leverage the automatic querying (with the automatic `supported` arg propagation). I see it quite beneficial even in RN world.

File: src/ya-comiste-react-native/src/entrypoints/ExploreMore.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native';
+import { fbt } from 'fbt';
+
+// type Props = {|
+//   +componentId: string,
+// |};
```
Comment (tbergq) [inline src/ya-comiste-react-native/src/entrypoints/ExploreMore.js:9]: Could also be removed?

Or you could leave it `type Props = {||}` and export the component as `export default (ExploreMore: ComponentType<Props>);`

File: src/ya-comiste-react-native/src/entrypoints/Profile.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+// @flow
+
+import * as React from 'react';
+import { Text, ScrollView } from 'react-native';
+import { fbt } from 'fbt';
+
+// type Props = {|
+//   +componentId: string,
+// |};
```
Comment (tbergq) [inline src/ya-comiste-react-native/src/entrypoints/Profile.js:9]: Same here

File: src/ya-comiste-react-native/src/entrypoints/Publish.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// @flow
+
+import * as React from 'react';
+import { Text, ScrollView } from 'react-native';
+import { fbt } from 'fbt';
+
+// type Props = {|
+//   +componentId: string,
+// |};
```
Comment (tbergq) [inline src/ya-comiste-react-native/src/entrypoints/Publish.js:9]: And here

File: src/ya-comiste-react-native/src/relay/QueryRenderer.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import * as React from 'react';
+import { SafeAreaView, StyleSheet, ActivityIndicator } from 'react-native';
+import { QueryRenderer as RelayQueryRenderer, type Variables } from 'react-relay';
+
+import relayEnvironment from './relayEnvironment';
+
+type Props = {|
+  +query: any, // TODO
```
Comment (tbergq) [inline src/ya-comiste-react-native/src/relay/QueryRenderer.js:10]: There should be a `GraphqlTaggedNode` or something exported from relay-runtime I think

**Review by tbergq — APPROVED** [review]

File: src/ya-comiste-react-native/src/BlockRenderer.js (hunk 4 above)
Comment (tbergq) [reply src/ya-comiste-react-native/src/BlockRenderer.js:20]: > You can still leverage the automatic querying (with the automatic supported arg propagation). I see it quite beneficial even in RN world.

Yeah, I didn't think about this, but I did a quick googling, and it seems using react.lazy should speed up start-up time for the apps so 👍

File: src/ya-comiste-react-native/src/relay/QueryRenderer.js (hunk 8 above)
Comment (mrtnzlml) [reply src/ya-comiste-react-native/src/relay/QueryRenderer.js]: fixed

**Pull request merged**